src/scripts/patch_ingame.lua

```diff
@@ -18,23 +18,78 @@ ScriptCB_DoFile("utility_functions2")
 --print("game_interface: Reading in custom strings")
 --ReadDataFile("v1.3patch_strings.lvl") -- where to put, root of'0'?
 
+-- trim "path\\to\\file.lvl" to "file"
+function trimToFileName(filePath)
+    local separatorIndex = 0
+    local lastSeparatorIndex = 0
 
+    -- Find the last occurrence of '\\'
+    local i = 1
+    while true do
+        local nextIndex = string.find(filePath, "\\", i, true)
+        if nextIndex then
+            lastSeparatorIndex = separatorIndex
+            separatorIndex = nextIndex
+            i = nextIndex + 1
+        else
+            break
+        end
+    end
+
+    local extensionIndex = string.find(filePath, "%.[^.]+$") -- Find the last occurrence of '.' followed by characters not including '.'
+    if extensionIndex then
+        filePath = string.sub(filePath, 1, extensionIndex - 1) -- Trim file extension
+    end
+
+    -- Extract substring after the last separator
+    if lastSeparatorIndex > 0 then
+        return string.sub(filePath, lastSeparatorIndex + 1)
+    else
+        return filePath
+    end
+end
 
 function RunUserScripts()
-    local maxScripts = 100
-    local i = nil
-    local patchDir = "..\\..\\addon\\0\\"
+
     print("RunUserScripts() Start")
-    for i = 0, maxScripts, 1 do
-        
-        if ScriptCB_IsFileExist(patchDir .. "user_script_" .. i .. ".lvl") ~= 0 then
-            print("game_interface: Found user_script_" .. i .. ".lvl")
-            
-            ReadDataFile(patchDir .. "user_script_" .. i .. ".lvl")
-            ScriptCB_DoFile("user_script_" .. i)
+
+    -- retrieve user scripts we saved in mission setup table
```

Comment:
Revisiting this functionality I do think we need to keep the 1.3 uop user script stuff.
But adding functionality like this will be really useful.
I would like to see the ability to select if the script gets run based on mission name also.

```LUA
-- path - the path to the script lvl file
-- name_of_script_to_execute - the name of the script inside of the script lvl to run
-- list_of_applicable_mission_names - list of mission names the script applies to; nil if it applies to everything.
function AddUserScript(path, name_of_script_to_execute, list_of_applicable_mission_names)

end
```

Reply:
You're right I should put the 1.3 way of adding user scripts back in too.

As for setting a user script to run based on that list, we still need a way to determine what mission is currently being played if we want that to work

Reply:
After further discussion, I do agree that the decision to run the script functionality is not necessary.
But I would like to see a 'remove' function as well (for use in menus, not so much in addme scripts).

```LUA
-- path - the path to the script lvl file
-- name_of_script_to_execute - the name of the script inside of the script lvl to run
function AddUserScript(path, name_of_script_to_execute )end

-- path - the path to the script lvl file
-- name_of_script_to_execute - the name of the script inside of the script lvl to run
function RemoveUserScript(path, name_of_script_to_execute )end
```